File: Data Structures & Algorithms/subsets/submission-2.cpp
class Solution {
public:

    void bt(int idx, vector<int>& nums, vector<int>& subset, vector<vector<int>>& res){
        res.push_back(subset);

        for(int i = idx; i < nums.size(); i++){
            subset.push_back(nums[i]);
            bt(i + 1, nums, subset, res);
            subset.pop_back();
        }


    }

    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> subset;

        bt(0, nums, subset, res);
        return res;

    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1] * len(nums)

        prefix = 1
        for i in range(len(nums)):
            res[i] *= prefix
            prefix *= nums[i]
        
        rightFix = 1
        for j in range(len(nums)-1, -1, -1):
            res[j] *= rightFix
            rightFix *= nums[j]

        return res
        